src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -516,6 +516,8 @@ class RichOption[T](val o: Option[T]) extends AnyVal {
   def contains(v: T): Boolean = o.isDefined && o.get == v
 
   override def toString: String = o.toString
+
+  def flatZip[U](other: Option[U]): Option[(T, U)] = o.flatMap { val1 => other.map(val2 => (val1, val2)) }
```

Comment:
Oddly, this is what Option.zip is supposed to be, except the return type in the code is wrong and it returns a 0- or 1-element list!  I'm not sure about the name.  I don't see where it flattens.  What about `liftedZip`?